sources/packages/api/src/esdc-integration/e-cert-integration/e-cert-file-handler.ts

```diff
@@ -203,22 +206,35 @@ export class ECertFileHandler extends ESDCFileHandler {
    * generate the record.
    * @returns e-Cert record.
    */
-  private createECertRecord(disbursement: DisbursementSchedule): ECertRecord {
+  private createECertRecord(
+    disbursement: DisbursementScheduleWithStopFullTimeBCFundingStatus,
+    offeringIntensity: OfferingIntensity,
+  ): ECertRecord {
     const now = new Date();
     const application = disbursement.studentAssessment.application;
     const addressInfo = application.student.contactInfo.address;
     const offering = application.currentAssessment.offering;
     const fieldOfStudy = getFieldOfStudyFromCIPCode(
       offering.educationProgram.cipCode,
     );
-    const awards = disbursement.disbursementValues.map(
-      (disbursementValue) =>
-        ({
-          valueType: disbursementValue.valueType,
-          valueCode: disbursementValue.valueCode,
-          valueAmount: disbursementValue.valueAmount,
-        } as Award),
-    );
+    const awards = [];
```

Comment:
Reason for converting the map to if, is to not populate the particular award if the award value type is BCGrant, when the student is having Stop Full Time BC Funding. When using map, undefined or null is getting populated. So thought this would be cleaner.

Reply:
use can `.filter()` it after `.map()`, to remove undefined or null. we do can similar example in our code

Reply:
Changed it to forEach as discussed.

Reply:
looking into it, @guru-aot  i think you can simply if like `    if (
      disbursement.stopFullTimeBCFunding &&
      OfferingIntensity.fullTime === offeringIntensity
    ){
      const awards = disbursement.disbursementValues.map((disbursementValue)=>{
        if (disbursementValue.valueType !== DisbursementValueType.BCGrant) {
          return  {
            valueType: disbursementValue.valueType,
            valueCode: disbursementValue.valueCode,
            valueAmount: disbursementValue.valueType === DisbursementValueType.BCLoan ? 0 : disbursementValue.valueAmount,
          } as Award
        }        
      })
    }`

Reply:
or something like this,    
```
 if (
      disbursement.stopFullTimeBCFunding &&
      OfferingIntensity.fullTime === offeringIntensity
    ){
      const awards = disbursement.disbursementValues.map((disbursementValue)=>(
            {
            valueType: disbursementValue.valueType,
            valueCode: disbursementValue.valueCode,
            valueAmount: disbursementValue.valueType === DisbursementValueType.BCLoan ? 0 : disbursementValue.valueAmount,
          } as Award
              
      )).filter(disbursementValue => disbursementValue.valueType !== DisbursementValueType.BCGrant)
    }
```

Reply:
I like it changed, thanks